Artifact recovery

hugokallander opened the pull request.

oeway committed:
Fix postgres and backup collection (#749)

fix postgres and backup collection

hugokallander committed:
Add initial recovery script

hugokallander committed:
Add progress artifact recovery script

hugokallander committed:
Add progress artifact recovery

oeway committed:
add script to backup

oeway committed:
Pin redis==5.2.0 (#751)

* bump version to 0.20.53; pin redis

* pin redis to 5.2.0

hugokallander committed:
Fix recover-artifacts.py

oeway committed:
fix docs

oeway committed:
Merge branch 'main' of github.com:amun-ai/hypha

oeway committed:
Fix s3 proxy (#752)

* fix s3 proxy

* format code

* pin websockets

* websockets>=14.0

* allow sharedarraybuffer

* Improve getting started

oeway committed:
remove cross-origin-opener-policy

oeway committed:
fix duplicated s3 query

oeway committed:
try to fix s3 proxy

oeway committed:
fix s3 proxy

oeway committed:
Fix s3 content encoding

oeway committed:
bump version

oeway committed:
Use improved asgiproxy

oeway committed:
Fix doc

oeway committed:
remove test s3 proxy

oeway committed:
fix test

oeway committed:
allow deleting artifact

oeway committed:
Support load balancing (#754)

* support load balancing

* Update workspace.py

oeway committed:
Rename service_load_balancing.md to service-load-balancing.md

hugokallander committed:
Merge branch 'main' into artifact-recovery

oeway reviewed (commented):
Hi, thanks for the PR, could you move the artifact-recovery scripts to utils folder instead of a separate one, there we already have the backup_database.py

Is there a command for dry-run the script without causing any damage, but check if the script working?

hugokallander commented:
Will do!

hugokallander committed:
Refactor move artifact recovery, fix extra artifact actions (mostly)

hugokallander committed:
Add dry-run option to artifact-recovery

hugokallander commented:
@oeway I moved the files to util/ and using only Claude I added the option --dry-run, but I haven't been able to test it fully because I lack the necessary environment variable values. Upon inspection, however, I think it looks pretty good

oeway commented:
OK, thanks! I think you can test it in the juptyer notebook we used to perform recovery, that has already env there.
Otherwise we put the PR on hold a bit until we tested it.
